src/library.rs

```diff
@@ -0,0 +1,44 @@
+use {
+    crate::{validator::Validator, ValidatorType},
+    std::{error::Error, result::Result},
+};
+
+// Holds all validators.
+// 1) Bootstrap
+// 2) Standard Validator -> One image for each validator (not implemented yet)
+// 3) RPC Node -> One image for each RPC node (not implemented yet)
+// 4) Clients -> Each client has its own image (not implemented yet)
+
+pub struct Library {
+    validators: Vec<Option<Validator>>,
```

Comment:
I agree it is a little weird. but it's basically used to store images and associated data. bootstrap, standard, rpc validators all use their own image for all validator types deployed. So a bootstrap, standard validator, and RPC would each only appear once in this Library. But for a Client, each client uses a different image, so that is why there is a vector of clients. This was the desired state with everything implemented: https://github.com/anza-xyz/validator-lab/pull/36/files#diff-47af127d675e3dc21b43c34222f42b9290334e88eb5c7ae0fc8b7539e3ecb794. But it may not be super clear.

i can change this to something like:
```
pub struct ValidatorImages { //struct name tbd
    bootstrap: Option<Validator>,
    standard: Option<Validator>,
    rpc: Option<Validator>,
    client: Vec<Validator>,
}
```
Note: we don't need a vector for `standard` or `rpc` since we only need one image per set of validators. Although i could change this if we want to store refactor a little more. 

Do something like:
```
pub struct Cluster { //struct name tbd
    bootstrap: Option<Validator>,
    standard: Vec<Validator>,
    rpc: Vec<Validator>,
    client: Vec<Validator>,
}
```

and then import the `kubernetes.rs` functionality here. so in `main` we could do something like:
```
cluster.create_and_deploy_validators()
```
^ and this would create all the labels, selectors, secrets, replicasets, etc for the non-bootstrap validators. and then would deploy all the validators. 

And can do the same for bootstrap, rpc, clients.

Reply:
I like that first option, it seems clearest to me at least.

And maybe we can split the difference on the name, and go with `ClusterImages` since the client isn't a validator? But I like both of the options :smile:

Reply:
ok cool sounds good! and yes true